org.eclipse.lemminx/src/test/java/org/eclipse/lemminx/extensions/contentmodel/XMLSchemaCompletionExtensionsTest.java

```diff
@@ -226,6 +233,57 @@ public void noNamespaceSchemaLocationCompletion() throws BadLocationException {
 				c("End with '</Configuration>'", "/Configuration>"),
 				c("End with '</ViewDefinitions>'", "/ViewDefinitions>"), c("End with '</View>'", "/View>"));
 	}
+	
+	@Test
+	public void customProtocolNamespaceSchemaLocationCompletionWhenCachingOn() throws BadLocationException {
+		
+		URL.setURLStreamHandlerFactory(new URLStreamHandlerFactory() {

```

Comment:
What about is other test implement too this static method setURLStreamHandlerFactory? I fear that we could have trouble, no?